lte/cloud/go/services/subscriberdb/storage/sub_proto.go

```diff
@@ -0,0 +1,335 @@
+// the token for the next page.
+func (l *SubProtoStore) GetPage(network string, token string, pageSize uint64) ([]*lte_protos.SubscriberData, string, error) {
+	lastIncludedSid := ""
+	if token != "" {
+		decoded, err := decodePageToken(token)
+		if err != nil {
+			return nil, "", err
+		}
+		lastIncludedSid = decoded.LastIncludedEntity
+	}
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		rows, err := l.builder.
+			Select(subProtoProtoCol).
+			From(subProtoTableName).
+			Where(squirrel.And{
+				squirrel.Eq{subProtoNidCol: network},
+				squirrel.Gt{subProtoSidCol: lastIncludedSid},
+			}).
+			OrderBy(subProtoSidCol).
+			Limit(pageSize).
+			RunWith(tx).
+			Query()
+		if err != nil {
+			return nil, errors.Wrapf(err, "get page for network %+v with token %+v", network, token)
+		}
+		return parseSubProtoRows(rows)
+	}
+	ret, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	if err != nil {
+		return nil, "", err
+	}
+	subProtos := ret.([]*lte_protos.SubscriberData)
+	nextPageToken, err := getNextPageToken(subProtos)
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "get next page token")
+	}
+	return subProtos, nextPageToken, nil
+}
+
+// GetByIDs returns an ordered list of subscriber protos with matching IDs.
+func (l *SubProtoStore) GetByIDs(network string, sids []string) ([]*lte_protos.SubscriberData, error) {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		rows, err := l.builder.
+			Select(subProtoProtoCol).
+			From(subProtoTableName).
+			Where(squirrel.And{
+				squirrel.Eq{subProtoNidCol: network},
+				squirrel.Eq{subProtoSidCol: sids},
+			}).
+			OrderBy(subProtoSidCol).
+			RunWith(tx).
+			Query()
+		if err != nil {
+			return nil, errors.Wrapf(err, "Get sub protos by ID for network %+v", network)
+		}
+		return parseSubProtoRows(rows)
+	}
+	ret, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	if err != nil {
+		return nil, err
+	}
+	subProtos := ret.([]*lte_protos.SubscriberData)
+	return subProtos, nil
+}
+
+// parseSubProtoRows scans db rows of serialized subscriber data, and returns a deserialized list
+// of subscriber protos.
+func parseSubProtoRows(rows *sql.Rows) ([]*lte_protos.SubscriberData, error) {
+	subProtos := []*lte_protos.SubscriberData{}
+	for rows.Next() {
+		subProtoMarshaled := []byte{}
+		err := rows.Scan(&subProtoMarshaled)
+		if err != nil {
+			return nil, errors.Wrap(err, "get sub protos from store, SQL row scan error")
+		}
+		subProto := &lte_protos.SubscriberData{}
+		err = proto.Unmarshal(subProtoMarshaled, subProto)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unmarshal sub protos from store")
+		}
+		subProtos = append(subProtos, subProto)
+	}
+	err := rows.Err()
+	if err != nil {
+		return nil, errors.Wrap(err, "get sub protos from store, SQL rows error")
+	}
+	return subProtos, nil
+}
+
+// getNextPageToken returns the next page token based on the lastIncludedEntity in the current page.
+func getNextPageToken(subProtos []*lte_protos.SubscriberData) (string, error) {
```

Comment:
These two page token fns probably do or should already exist in configurator -- can we just make those public (or move these to configurator and make them public) and then reuse them in both places